scrapfly/extraction_config.py

```diff
@@ -203,3 +235,44 @@ def from_dict(extraction_config_dict: Dict) -> 'ExtractionConfig':
             webhook=webhook,
             raise_on_upstream_error=raise_on_upstream_error
         )
+
+
+def detect_compression_format(data):
+    """
+    Detects the compression type of the given data.
+
+    Args:
+        data: The compressed data as bytes.
+
+    Returns:
+        The name of the compression type ("gzip", "zstd", "deflate", "unknown").
+    """
+
+    if len(data) < 2:
+        return 'unknown'
+
+    # gzip 
+    if data[0] == 0x1f and data[1] == 0x8b:
+        return 'gzip'
```

Comment:
We can directly use the CompressionFormatEnum, remove the translation  between string to enum and add a return type accordingly